server/hotrod/src/main/java/org/infinispan/server/hotrod/BaseDecoder.java

```diff
@@ -62,11 +63,15 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
    @Override
    public void channelWritabilityChanged(ChannelHandlerContext ctx) throws Exception {
       super.channelWritabilityChanged(ctx);
+      Channel channel = ctx.channel();
+      boolean writeable = channel.isWritable();
       if (trace) {
-         log.tracef("Channel %s writability changed", ctx.channel());
+         log.tracef("Channel %s writability changed to %b", channel, writeable);
+      }
+      if (writeable) {
+         server.getClientListenerRegistry().findAndWriteEvents(channel);
+         server.getClientCounterNotificationManager().channelActive(channel);
```

Comment:
Maybe we should check that the channel is still writable after sending the events?

Reply:
I meant it's much more likely that the channel is no longer active after we wrote some events than right after we got the notification that the channel is active. Unless `super.channelWritabilityChanged(ctx)` can also cause some writes to the channel?

Reply:
Not sure if this is intermingled with the other comment chain now? But yes I agree that the channel may not be writeable most likely after exhausting all 100+ elements from the event queue. But I don't understand what change you are proposing here though. If the channel is no longer writeable after writing the events, we just wait until we get the callback again telling us the channel is writeable to continue writing more.